server.js

```diff
@@ -16,62 +11,31 @@ app.get("/index", (req, res) => {
 });
 
 app.get("/generateraceAPI", (req, res) => {
-  data = {
-    hash: crypto.randomBytes(20).toString("hex"),
-  };
-  res.json(data);
+  let hash = helper.generateNewHash();
+  raceRooms.get(hash, (err, room) => {
+    if (room == null) {
+      data = {
+        hash: hash,
+      };
+      res.json(data);
+    } else {
+      generateNewHash(previoushash);
```

Comment:
hash not previousHash.

Reply:
But this else condition has no effect as:
1. `generateNewHash` is an undefined, it is `helper.generateNewHash`. It hasn't thrown an error cause this condition was never met (but a language like Go would've thrown 👅 )
2. Even if it returns an hash, it is not being reinserted in redis. If you want to have recursion it should be in this function, as redis will tell whether key is existing or not so no need for previous check.

Reply:
But recursion is not needed if always know we generate a unique hash, let's try to do this and explore unique parameters to the hash function.

Reply:
Lol yeah made some errors, Will explain the else part in the call